### ISSUE TITLE
LDAP groups should be in a different OU than users

### ISSUE THREAD

**Issue by jamshid** [bitnami/containers#19716]
### Name and Version

bitnami/openldap:2.6

### What steps will reproduce the bug?

Create an LDAP container with a user and group, why are groups in the same OU as users?

### What is the expected behavior?

Groups should be under a different ou, not users or people right? E.g.
https://serverfault.com/questions/87079/openldap-posix-groups-account
```
# User primary group
dn: cn=ussergroup,ou=groups,dc=me,dc=com
...
# User account
dn: uid=user,ou=users,dc=me,dc=com
```
Please add a LDAP_GROUP_OU similar to LDAP_USER_DC and use that in the template.
Btw LDAP_USER_DC should probably be renamed to "LDAP_USER_OU", since it defaults to ou=users not a dc?

### What do you see instead?

`dn: cn=mygroup,ou=users,dc=example,dc=org`

### Additional information

this originally came up in https://github.com/bitnami/bitnami-docker-openldap/issues/2

**Comment by carrodher** [comment]
Thanks for reporting this issue. Would you like to contribute by creating a PR to solve the issue? The Bitnami team will be happy to review it and provide feedback. [Here](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md) you can find the contributing guidelines.

**Comment by jamshid** [comment]
@dgomezleon @barsikus007 this issue was not fixed. The env var was changed but the group created is still in `ou=users` instead of `ou=groups`. Adding `-e LDAP_GROUP_OU=groups` does not help.

`docker run -d --name openldap -p 1389:1389  bitnami/openldap:latest` 

```
$ curl 'ldap://localhost:1389/dc=example,dc=org?*?sub?(objectclass=*)'
DN: ou=users,dc=example,dc=org
	objectClass: organizationalUnit

	ou: users

DN: cn=user01,ou=users,dc=example,dc=org
...
DN: cn=readers,ou=users,dc=example,dc=org
	cn: readers

	objectClass: groupOfNames

	member: cn=user01,ou=users,dc=example,dc=org
	member: cn=user02,ou=users,dc=example,dc=org
```
There should be a
`DN: ou=groups,dc=example,dc=org`
and that's where the `readers` group should live:
`DN: cn=readers,ou=groups,dc=example,dc=org`

**Comment by jamshid** [comment]
Here's the bug -- group creation is still referencing LDAP_USER_OU:
https://github.com/bitnami/containers/blob/278116ba7f5c44373c5537f3ec4cec17fa533c61/bitnami/openldap/2.6/debian-12/rootfs/opt/bitnami/scripts/libopenldap.sh#L579

I made a PR @dgomezleon @barsikus007 lmk if any problems:
https://github.com/bitnami/containers/pull/77099

**Comment by jamshid** [comment]
Thanks @alvneiayu for merging the PR. Any idea when a new bitnami/openldap image build (2.6.10?) will be available?

**Comment by carrodher** [comment]
The latest tags were pushed some hours ago, see https://hub.docker.com/r/bitnami/openldap/tags

**Comment by jamshid** [comment]
Thanks @carrodher but is there something else that has to be done to get the change into `bitnami/openldap:latest`? It's still using the old code:

```
% docker pull bitnami/openldap:latest
latest: Pulling from bitnami/openldap
Digest: sha256:e1929afbb392b76baa1c4508bdaff613423b478620a7130d28c4cd9f39fef084
Status: Image is up to date for bitnami/openldap:latest
docker.io/bitnami/openldap:latest

% docker run bitnami/openldap:latest fgrep 'dn: ${LDAP_GROUP/' opt/bitnami/scripts/libopenldap.sh

dn: ${LDAP_GROUP/#/cn=},${LDAP_USER_OU/#/ou=},${LDAP_ROOT}
```

**Comment by jamshid** [comment]
Ah this bitnami-bot commit reverted my change. IDK why. Just asked on the commit. 

Maybe it is rebuilding 2.6.9 for some reason and doesn't want to introduce a change. But seems it shouldn't break/revert what's on main.

https://github.com/bitnami/containers/commit/f05a5976bde823582ed29ce1d3a7bb8a7c60d9bd
```
commit f05a5976bde823582ed29ce1d3a7bb8a7c60d9bd
Author: Bitnami Bot <bitnami-bot@vmware.com>
Date:   Mon Feb 10 20:31:38 2025 +0100

    [bitnami/openldap] Release openldap-2.6.9-debian-12-r4 (#77147)
    
    Signed-off-by: Bitnami Bot <bitnami-bot@vmware.com>

commit d8158d6a510a222cbb5f36bd4abf3b378343e1e6
```

**Comment by alvneiayu** [comment]
Hi @jamshid 

No worries, I will fix it.

Sorry for the inconveniences

Alvaro

**Comment by jamshid** [comment]
Thanks @alvneiayu! I confirmed `bitnami/openldap:latest` is correct. Hopefully it gets a release tag like `2.6.10` soon so users can pin to it.

**Comment by jamshid** [comment]
Thanks for putting the fix in 2.6.9. 
But I'm surprised you changed an existing image's behavior rather than publishing a new 2.6.10. 

```
% docker run --pull always bitnami/openldap:2.6.9 fgrep 'dn: ${LDAP_GROUP/' opt/bitnami/scripts/libopenldap.sh
2.6.9: Pulling from bitnami/openldap
Digest: sha256:63d31cdaa455c85abeb43d7d0b3bd1afcf0f671b602fe960f02b6a4b713f9e2a
Status: Downloaded newer image for bitnami/openldap:2.6.9

dn: ${LDAP_GROUP/#/cn=},${LDAP_GROUP_OU/#/ou=},${LDAP_ROOT}
```

Anyone relying on the previous behavior can pin to 2.6.8.
```
% docker run --pull always bitnami/openldap:2.6.8 fgrep 'dn: ${LDAP_GROUP/' opt/bitnami/scripts/libopenldap.sh
2.6.8: Pulling from bitnami/openldap
555a9ecc9e42: Pull complete 
Digest: sha256:ef16af5f5565168fd77b3f728b3e4dc51538f80ea4c736af2c288eb55900b5c6
Status: Downloaded newer image for bitnami/openldap:2.6.8

dn: ${LDAP_GROUP/#/cn=},${LDAP_USER_DC/#/ou=},${LDAP_ROOT}
```
